File: JavascriptAlgorithm/LinkedList/hasCycle-141.js
var hasCycle = function (head) {
  /* let slow = head,
    fast = head.next;
  while (slow !== fast) {
    if (slow === null || fast === null) {
      return false;
    }
    slow = slow.next;
    fast = fast.next !== null ? fast.next.next : null;
  }
  return true; */
  if (head === null) {
    return false;
  }
  let slow = head,
    fast = head;
  while (fast !== null && fast.next !== null) {
    slow = slow.next;
    fast = fast.next.next;
    if (slow === fast) {
      return true;
    }
  }
  return false;
};
export { hasCycle };

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/ThirdRound/Chapter03/hasCycle_141.java
package algo.tzashinorpu.ThirdRound.Chapter03;

import algo.tzashinorpu.ThirdRound.ListNode;

public class hasCycle_141 {
    public boolean hasCycle(ListNode head) {
        ListNode slow = head;
        ListNode fast = head;
        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
            if (slow == fast) {
                return true;
            }
        }
        return false;
    }
}
